ocaml/xapi/xapi_observer_components.ml

```diff
@@ -54,8 +58,34 @@ let assert_valid_components components =
   with Unsupported_Component component ->
     raise Api_errors.(Server_error (invalid_value, ["component"; component]))
 
+let filter_out_exp_components components =
+  let experimental_components_list =
+    match !Xapi_globs.observer_experimental_components with
+    | "" ->
+        []
+    | lst ->
+        lst
+        |> String.split_on_char ','
+        |> List.filter_map (fun elt ->
+               try Some (of_string elt)
+               with Unsupported_Component elt ->
+                 debug "Unsupported component in xapi.conf: %s" elt ;
+                 None
+           )
+  in
+  components
+  |> List.filter (fun component ->
+         not (List.mem component experimental_components_list)
```

Comment:
this would be more efficient as the intersection of 2 sets, than lists.
Even if we keep the list we could also compute the experimental set just once on startup right when setting it (I think you can specify a custom setter, use `Arg.String` instead of `Arg.Set_string`).

Reply:
I like the idea of a custom setter using Arg.String

Reply:
@edwintorok I made a custom setter and used sets. https://github.com/xapi-project/xen-api/pull/5428/commits/f02d70c942fa0df497d32df613c070218405d20a